ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -1840,20 +1854,27 @@ def test_opensslchainvalidation_ipa_ca_cert(self, replace_ipa_chain):
         Test for IPAOpenSSLChainValidation when /etc/ipa/ca.crt
         contains IPA CA cert but not the external CA
         """
+        version = tasks.get_healthcheck_version(self.master)
         error_msg = "Certificate validation for {key} failed: {reason}"
+        error_msg_40_txt = (
+            "CN = Certificate Authority\nerror 2 at 1 depth "
+            "lookup: unable to get issuer certificate\n"
+        )
         returncode, data = run_healthcheck(
             self.master,
             "ipahealthcheck.ipa.certs",
             "IPAOpenSSLChainValidation",
         )
         assert returncode == 1
         for check in data:
-            if check["kw"]["key"] == paths.HTTPD_CERT_FILE:
-                assert check["result"] == "ERROR"
-                assert error_msg in check["kw"]["msg"]
-            elif check["kw"]["key"] == paths.RA_AGENT_PEM:
-                assert check["result"] == "ERROR"
-                assert error_msg in check["kw"]["msg"]
+            assert check["result"] == "ERROR"
+            if parse_version(version) >= parse_version("0.6"):
+                if check["kw"]["key"] == paths.HTTPD_CERT_FILE:
+                    assert error_msg in check["kw"]["msg"]
+                elif check["kw"]["key"] == paths.RA_AGENT_PEM:
+                    assert error_msg in check["kw"]["msg"]
+            else:
+                assert error_msg_40_txt in check["kw"]["msg"]
```

Comment:
reason should be defined in 0.6 and I'd guess is the same value as error_msg_40_txt. It might be worthwhile to rename the variable to error_reason or something and compare that against reasin when >= 0.6 and msg when not.

Reply:
modified the variable name to error_reason and also comared that against reason.